### PR TITLE
Adds unit + integration testing to UV

### LINKED ISSUE #575: Set up puppeteer in UV to use for integration tests

### PR DESCRIPTION
**Pull request by mejackreed**
Closes #575 
In the spirit of #568 but doesn't use mocha
Also related to #567

`__tests_/test.js` holds the previous mocha test and a new proof of concept integration test. Unit tests can also be added in the project either in their respective module directories but there will be some additional setup needed for this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,11 +46,15 @@
     "grunt-text-replace": "0.3.7",
     "grunt-ts": "6.0.0-beta.11",
     "hooker": "0.2.3",
+    "http-server": "^0.11.1",
+    "jest": "^22.4.3",
+    "jest-puppeteer": "^2.2.0",
     "json-schema-generator": "edsilv/json-schema-generator",
     "less": "2.4.0",
     "lodash": "4.17.4",
     "mocha": "4.0.1",
```
Comment (edsilv) [inline package.json]: remove mocha?

File: package.json [hunk 2]
```diff
@@ -44,11 +46,15 @@
...
     "ncp": "2.0.0",
+    "puppeteer": "^1.2.0",
     "selenium-webdriver": "3.6.0",
```
Comment (edsilv) [inline package.json]: remove selenium?

File: package.json (hunk 1 above)
Comment (mejackreed) [reply package.json]: Nice catch.. removed.

File: package.json (hunk 2 above)
Comment (mejackreed) [reply package.json]: Nice catch.. removed.

**Pull request merged**